packages/oui-datagrid/src/datagrid.controller.js

```diff
@@ -80,16 +80,45 @@ export default class DatagridController {
         const builtColumns = this.buildColumns();
         this.previousRows = this.columns;
 
-        if (this.rowsLoader) {
-            this.paging = this.ouiDatagridPaging.createRemote(this.columns, builtColumns.currentSorting, this.pageSize, this.rowLoader, this.rowsLoader);
-            this.refreshData(() => this.paging.setOffset(1));
-        } else {
-            this.paging = this.ouiDatagridPaging.createLocal(this.columns, builtColumns.currentSorting, this.pageSize, this.rowLoader, this.rows);
-
-            if (this.rows) {
-                this.refreshData(() => this.paging.setRows(this.rows));
+        const refreshDatagrid = () => {
+            if (this.rowsLoader) {
+                this.paging = this.ouiDatagridPaging.createRemote(this.columns, builtColumns.currentSorting, this.pageSize, this.rowLoader, this.rowsLoader);
+                this.refreshData(() => this.paging.setOffset(1));
+            } else {
+                this.paging = this.ouiDatagridPaging.createLocal(this.columns, builtColumns.currentSorting, this.pageSize, this.rowLoader, this.rows);
+
+                if (this.rows) {
+                    this.refreshData(() => this.paging.setRows(this.rows));
+                }
             }
-        }
+        };
+
+        this.$scope.$on("datagrid-refresh", () => {
+            if (!this.refreshDatagridPromise) {
+                refreshDatagrid();
+            }
+        });
+        refreshDatagrid();
+
+        this.$scope.$on("datagrid-update-rows", (ev, updater) => {
+            const doUpdate = () => {
+                if (angular.isFunction(updater)) {
```

Comment:
Don't you have `_` instead ? We should stop switching between `angular` and `_`

Reply:
use `angular` built-in methods otherwise lodash.

Reply:
Why ? It's the first time I've heard about that, is this a convention ?